fetcher.go

```diff
@@ -137,7 +151,21 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 				fetchDurationPerCarPerPeerSuccessMetric.WithLabelValues(cacheStatus).Observe(float64(response_success_end.Sub(start).Milliseconds()))
 			}
 
-			updateSuccessServerTimingMetrics(respHeader.Values(servertiming.HeaderKey), resourceType, isCacheHit, durationMs, ttfbMs, received)
+			latencyMillis, speedBytesPerMs := updateSuccessServerTimingMetrics(respHeader.Values(servertiming.HeaderKey), resourceType, isCacheHit, durationMs, ttfbMs, received)
+
+			if latencyMillis > 0 && speedBytesPerMs > 0 {
+				p.lk.Lock()
```

Comment:
are we going to get in trouble if we lock full pool multiple times on every request?

Reply:
done.

Reply:
what are you going to use to see if it's a problem?